src/rdbms/mongoose_rdbms.erl

```diff
@@ -542,18 +542,18 @@ inner_transaction(F, _State) ->
 outer_transaction(F, NRestarts, _Reason, State) ->
     sql_query_internal(rdbms_queries:begin_trans(), State),
     put_state(State),
-    Result = try
-                 F()
-             catch
-                 throw:ThrowResult ->
-                     ThrowResult;
-                 Class:Other ->
-                     Stacktrace = erlang:get_stacktrace(),
-                     ?ERROR_MSG("issue=outer_transaction_failed "
-                                "reason=~p:~p stacktrace=~1000p",
-                                [Class, Other, Stacktrace]),
-                     {'EXIT', Other}
-          end,
+    {Result, StackTrace} =
+        try
+            {F(), []}
+        catch
+            throw:ThrowResult:S ->
+                {ThrowResult, S};
+            Class:Other:Stacktrace ->
+                ?ERROR_MSG("issue=outer_transaction_failed "
+                           "reason=~p:~p stacktrace=~1000p",
+                           [Class, Other, Stacktrace]),
+                {{'EXIT', Other},Stacktrace}
+        end,
```

Comment:
A cosmetic thing. I think readability of this block could be improved a bit by moving they `try ... catch` clause to a dedicated function like `apply_transaction_function`. What do you think?